include/mycroft/file.h

```diff
@@ -5,48 +5,73 @@
 extern "C" {
 #endif
 
+/* #include <stdbool.h> */
+#include <stdio.h>
 #include <stdint.h>
 #include <mycroft/template.h>
+#include <mycroft/error.h>
 
 /** Size used by internal file functions */
-typedef long int fsize_t;
+typedef off_t fsize_t;
 
 typedef struct mc_file_t mc_file_t;
 
 /* Util functions */
-int file_exists(const char* file);
+int mc_file_access(const char* file_name);
 
 /* Init/free functions */
 mc_file_t* mc_file_create();
-mc_file_t* mc_file_open(const char* file_name);
-int mc_file_close(mc_file_t* file);
-int mc_file_reload(mc_file_t* file);
+mc_file_t* mc_file_open(const char* file_name, mc_error_t* err);
```

Comment:
Returning NULL should be sufficient indicator of error for this function, adding in an additional parameter for returning error codes isn't necessary.